discovery-infra/test_infra/controllers/node_controllers/terraform_controller.py

```diff
@@ -43,10 +43,10 @@ def _get_random_name(cls):
 
     # TODO move all those to conftest and pass it as kwargs
     def _terraform_params(self, **kwargs):
-        params = {"libvirt_worker_memory": kwargs.get('worker_memory'),
-                  "libvirt_master_memory": kwargs.get('master_memory', 16984),
-                  "libvirt_worker_vcpu": kwargs.get("worker_vcpu", 4),
-                  "libvirt_master_vcpu": kwargs.get("master_vcpu", 4),
+        params = {"libvirt_worker_memory": self._resource_param(self, 'worker_memory', kwargs),
```

Comment:
Please reuse the non-kwargs version `self._resource_param` from the `start_discovery`. The summing logic should only be implemented once. Any differences specialized for simpler terraform `kwargs` access can be done in a thin wrapper.

You might want to make the `start_discovery` version of `self._resource_param` receive the tables as a parameter, making it more flexible for use by the terraform wrapper.